sycl/plugins/opencl/CMakeLists.txt

```diff
@@ -26,14 +33,37 @@ set_target_properties(pi_opencl PROPERTIES LINKER_LANGUAGE CXX)
 #preprocessor definitions for compiling a target's sources. We do not need it for pi_opencl
 target_include_directories(pi_opencl PRIVATE "${sycl_inc_dir}")
 
-#link pi_opencl with OpenCL headers and ICD Loader.
-target_link_libraries( pi_opencl
+if (MSVC)
```

Comment:
What is the "shlwapi"? Are we going to get the same affect on Windows that only pi* symbols are exported from the plugin?

Reply:
> What is the "shlwapi"? Are we going to get the same affect on Windows that only pi* symbols are exported from the plugin?

shlwapi is the common library included to avoid unresolved symbols error when compiling with Visual Studio.
I copied this line from CMakelists.txt for libopencl.so.
Yes, on Windows, we use __declspec(dllexport) for pi* symbols which indicate exported.
So, we export only pi* symbols from pi_opencl.cpp.

Reply:
> I copied this line from CMakelists.txt for libopencl.so.

Why do we need to follow what libopencl.so is doing? Can you show what are the unresolved symbols, and how this library fixes them?